mqtt-format/src/v3/packet.rs

```diff
@@ -214,13 +216,37 @@ impl<'message> MPacket<'message> {
                     .await?;
             }
             MPacket::Publish {
-                dup: _,
-                qos: _,
-                retain: _,
-                topic_name: _,
-                id: _,
-                payload: _,
-            } => todo!(),
+                dup,
+                qos,
+                retain,
+                topic_name,
+                id,
+                payload,
+            } => {
+                let packet_type = 0b0011_0000;
+                let dup_mask = if *dup { 0b0000_1000 } else { 0 };
+                let qos_mask = qos.to_byte() << 1;
+                let retain_mask = *retain as u8;
```

Comment:
Some comment here would be nice, as this is very "magic constant"y.

Reply:
I think the variable names are very clear?